core/templates/dev/head/components/top_navigation_bar/TopNavigationBarDirective.js

```diff
@@ -139,9 +140,10 @@ oppia.directive('topNavigationBar', [
             $('nav a').blur();
           };
           $scope.closeSubmenu = function(evt) {
-            $scope.activeMenuName = '';
-            angular.element(evt.currentTarget).closest('li')
-              .find('a').blur();
+            NavigationService.closeSubmenu(evt, menuName);
```

Comment:
Why the parameter `menuName` here? 

Check this function in NavigationService.js, it's not taking in the parameter `menuName`. Remove this and it will fix your issue.

Reply:
Are you still getting the same error ?

Reply:
Yes.

Reply:
In navigationservice.onMenuKeypress function the else conditions error is being thrown.

Reply:
Umm, interesting. I'm not getting the same error now. Could you upload a video for the same (just make sure you have made these changes) ?

(I'm getting a different error now)

Reply:
Sure :)
[Drive Link](https://drive.google.com/open?id=1wFFtKNj-3Cjtj5AtJBEwCqTruulwtsQ5)

Reply:
Thanks for the link!

In html templates if you have to access the rootScope variables than `$root` is used. So code will look like `$root.ACTION_OPEN`, `$root.ACTION_CLOSE` etc.

Could you try this?

Reply:
If I replace all the $rootScope with $root, I get this

<img width="1440" alt="screenshot 2018-12-25 at 12 23 05 am" src="https://user-images.githubusercontent.com/35570715/50405487-44804500-07db-11e9-93dd-adbc44e39691.png">

<img width="1440" alt="screenshot 2018-12-25 at 12 22 57 am" src="https://user-images.githubusercontent.com/35570715/50405490-48ac6280-07db-11e9-8584-d12b9807ed06.png">

Reply:
I meant to say in html templates, not in js files. 

Replace `ACTION_OPEN` with `$root.ACTION_OPEN` in html files.

Reply:
It works. Thanks a lot :). I will push the changes now.

Reply:
Great!

Reply:
@bansalnitish PTAL